overlord/devicestate/devicestate_remodel_test.go

```diff
@@ -909,13 +909,24 @@ func (s *deviceMgrRemodelSuite) TestReregRemodelClashAnyChange(c *C) {
 		"required-snaps": []interface{}{"new-required-snap-1", "new-required-snap-2"},
 		"revision":       "1",
 	})
+	s.newFakeStore = func(devBE storecontext.DeviceBackend) snapstate.StoreService {
+		mod, err := devBE.Model()
+		c.Check(err, IsNil)
+		if err == nil {
```

Comment:
doen't the check on the line above ensure that this condition will be hit?

Reply:
No, c.Check() does not stop a test execution if the condition is false, that would be c.Assert(). OTOH, calling an assert in callback that can run under taskrunner (i.e. in a separate goroutine and cause a panic there), is problematic and often makes the failing test either hang or produce a backtrace that obscures the original failure.

Also, on a closed look, with the changed conflict check, the callback would not be called at all. I'll push a patch